src/overlays/actors/ovl_En_Rl/z_en_rl.c

```diff
@@ -9,81 +16,379 @@ void EnRl_Destroy(Actor* thisx, GlobalContext* globalCtx);
+        this->action = 3;
+    }
+}
+
+void func_80AE7798(EnRl* this, GlobalContext* globalCtx) {
+    func_80AE7668(this, globalCtx);
+}
+
+void func_80AE77B8(EnRl* this, GlobalContext* globalCtx) {
+    func_80AE744C(this, globalCtx);
+    func_80AE7494(this);
+    func_80AE72D0(this);
+    func_80AE7698(this, globalCtx);
+}
+
+void func_80AE77F8(EnRl* this, GlobalContext* globalCtx) {
+    s32 temp;
+
+    func_80AE744C(this, globalCtx);
+    temp = func_80AE7494(this);
+    func_80AE72D0(this);
+    func_80AE772C(this, temp);
+}
+
+void func_80AE7838(EnRl* this, GlobalContext* globalCtx) {
+    func_80AE744C(this, globalCtx);
+    func_80AE7494(this);
+    func_80AE72D0(this);
+    func_80AE7590(this, globalCtx);
+}
+
+void func_80AE7878(EnRl* this, GlobalContext* globalCtx) {
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_06007B38, &D_06000A3C, 0, 0, 0);
+    this->action = 4;
+    this->actor.shape.unk_14 = 0;
+}
+
+void func_80AE78D4(EnRl* this, GlobalContext* globalCtx) {
+    Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_DEMO_6K, this->actor.posRot.pos.x,
+                       kREG(18) + 22.0f + this->actor.posRot.pos.y, this->actor.posRot.pos.z, 0, 0, 0, 5);
+}
+
+void func_80AE7954(EnRl* this, GlobalContext* globalCtx) {
+    if (func_80AE74B4(this, globalCtx, 4, 0)) {
+        this->action = 5;
+        this->drawConfig = 2;
+        this->alpha = 0;
+        this->actor.shape.unk_14 = 0;
+        this->unk_19C = 0.0f;
+    }
+}
+
+void func_80AE79A4(EnRl* this, GlobalContext* globalCtx) {
+    f32* unk_19C = &this->unk_19C;
+    s32 alpha = 255;
+
+    if (func_80AE74B4(this, globalCtx, 4, 0)) {
+        *unk_19C += 1.0f;
+        if (*unk_19C >= kREG(5) + 10.0f) {
+            this->action = 7;
+            this->drawConfig = 1;
+            *unk_19C = kREG(5) + 10.0f;
+            this->alpha = alpha;
+            this->actor.shape.unk_14 = alpha;
+            return;
+        }
+    } else {
+        *unk_19C -= 1.0f;
+        if (*unk_19C <= 0.0f) {
+            this->action = 4;
+            this->drawConfig = 0;
+            *unk_19C = 0.0f;
+            this->alpha = 0;
+            this->actor.shape.unk_14 = 0;
+            return;
+        }
+    }
+    alpha = (*unk_19C / (kREG(5) + 10.0f)) * 255.0f;
+    this->alpha = alpha;
+    this->actor.shape.unk_14 = alpha;
+}
+
+void func_80AE7AF8(EnRl* this, GlobalContext* globalCtx) {
+    if (func_80AE74B4(this, globalCtx, 3, 0)) {
+        SkelAnime_ChangeAnim(&this->skelAnime, &D_0600040C, 1.0f, 0.0f,
+                             SkelAnime_GetFrameCount(&D_0600040C.genericHeader), 2, -8.0f);
+        this->action = 6;
+    } else if (func_80AE74FC(this, globalCtx, 4, 0)) {
+        this->action = 5;
+        this->drawConfig = 2;
+        this->unk_19C = kREG(5) + 10.0f;
+        this->alpha = 255;
+        if (this->unk_1A4 == 0) {
+            func_80AE78D4(this, globalCtx);
+            this->unk_1A4 = 1;
```

Comment:
`func_80AE78D4` spawns `ACTOR_DEMO_6K` which the cloudmodding wiki described as "Sages, Balls of Light (Cutscene)", I'm not sure if this refers to when the sages transform into light balls to jail ganon or something.
I think `unk_1A4` can still be safely renamed to something like `isLightBallSpawned` maybe? bad name